core/shared/src/main/scala/zio/config/PropertyTree.scala

```diff
@@ -116,12 +117,12 @@ sealed trait PropertyTree[+K, +V] { self =>
    * @param that
    * @return
    */
-  final def merge[K1 >: K, V1 >: V](that: PropertyTree[K1, V1]): List[PropertyTree[K1, V1]] =
+  final def merge[K1 >: K, V1 >: V](that: PropertyTree[K1, V1]): NonEmptyList[PropertyTree[K1, V1]] =
     (self, that) match {
-      case (Sequence(l), Sequence(r))           => singleton(Sequence(l ++ r))
+      case (Sequence(l), Sequence(r))           => NonEmptyList(Sequence(l ++ r))
       case (l: Record[K, V], r: Record[K1, V1]) =>
         (l.value.keySet ++ r.value.keySet)
-          .foldLeft(List[Map[K1, PropertyTree[K1, V1]]](Map.empty)) { case (acc, k) =>
```

Comment:
Hmm. Just for NonEmptyList, we are not yet in a position to depend on Zio-prelude. A temporary set up for NonEmptyList in prelude package is a nice idea. In future, I can see zio-config depending on zio-prelude though.

Reply:
@afsalthaj I could implement a minimal `NEL` based of `zio-prelude` for now. Let me know where I should place the new `NEL`.

Reply:
@afsalthaj I've just realised that there's a `NonEmptyChunk` that does not depend on `zio-prelude`. I'm not sure if it affects perfomance switch from `List` to `Chunk`.

Reply:
yeah good point. It’s always good to move from List to Chunk.

Reply:
@afsalthaj I've switched to `NonEmptyChunk` and dropped the `zio-prelude` dependency.

Reply:
Fantastic. I am facing some scala3 issues in master branch. So might delay the release.